File: src/leetcode/easy/MissingNumber.java
package leetcode.easy;

/**
 * Question: https://leetcode.com/problems/missing-number/description/
 */
public class MissingNumber {
    public int missingNumber(int[] nums) {
        int sum = 0, n = nums.length;
        for (int num : nums) sum += num;
        return ((n * (n + 1)) / 2) - sum;
    }
}

File: javascript/easy/missingNumber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    let n = nums.length, expectedSum = parseInt(((n+1) * n)/2), sum = 0;
    for(let i = 0; i < n; i++) {
        sum += nums[i];
    }
    return expectedSum - sum;
};
